sgt-back-end/index.js

```diff
@@ -0,0 +1,159 @@
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades', (req, res) => {
+  const sql = 'select * from "grades"';
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      if (!grades) {
+        res.status(500).json({ error: 'An unexpected error occurred.' });
+      } else {
+        res.status(200).json(grades);
+      }
+    })
+    .catch(err => {
+      console.error(err.message);
+      res.status(500).json({ error: 'An unexpected error occurred.' });
+    });
+});
+
+app.get('/api/grades/:gradeId', (req, res) => {
+  if (!(req.params.gradeId)) {
+    res.status(400).json({ error: 'gradeId is a required fields' });
+    return;
+  }
+  const gradeId = parseInt(req.params.gradeId);
+  if (!(gradeId.toString().length === req.params.gradeId.length && gradeId > 0)) {
+    res.status(400).json({ error: 'gradeId must be a positive integer' });
+    return;
+  }
+
+  const sql = `
+      select *
+      from "grades"
+      where "gradeId" = $1
+      `;
+  const params = [req.params.gradeId];
+  db.query(sql, params)
+    .then(result => {
+      const studentGrade = result.rows[0];
+      if (!studentGrade) {
+        res.status(404).json({ error: `cannot find grade with id ${gradeId}` });
+      } else {
+        res.status(200).json(studentGrade);
+      }
+    })
+    .catch(err => {
+      console.error(err.message);
+      res.status(500).json({ error: 'An unexpected error occurred.' });
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  if (!(req.body.name && req.body.course && req.body.score)) {
+    res.status(400).json({ error: 'content, course, & score are all required fields' });
+    return;
+  }
+  const score = parseInt(req.body.score);
+  if (!(score.toString().length === req.body.score.length && score > 0)) {
+    res.status(400).json({ error: 'score must be a positive integer' });
+    return;
+  }
+
+  const sql = `
+      insert into
+      "grades" ("name", "course", "score")
+      values ($1, $2, $3)
+      returning *
+      `;
+  const params = [req.body.name, req.body.course, req.body.score];
+
+  db.query(sql, params)
+    .then(result => {
+      const studentGrade = result.rows[0];
+      if (!studentGrade) {
+        res.status(500).json({ error: 'An unexpected error occurred.' });
+      } else {
+        res.status(201).json(studentGrade);
+      }
+    })
+    .catch(err => {
+      console.error(err.message);
+      res.status(500).json({ error: 'An unexpected error occurred.' });
+    });
+});
+
+app.put('/api/grades/:gradeId', (req, res) => {
+  if (!(req.body.name && req.body.course && req.body.score)) {
+    res.status(400).json({ error: 'name, course, & score are required fields' });
+    return;
+  }
+  const gradeId = parseInt(req.params.gradeId);
+  const score = parseInt(req.body.score);
+  if (!((score.toString().length === req.body.score.length && score > 0) &&
```

Comment:
i don't really understand why these `length` comparisons are necessary. does your code work as intended if you remove all of these?
```js
score.toString().length === req.body.score.length
```

Reply:
These lines make sure the numbers being passed are only integers.
Without the .lengths, someone can pass "api/grades/5LOL"
The ID would be 5.

To make sure this gets avoided, I'm comparing the length of the return of parseInt() to the length of the initial value.

But I can remove these lines if you believe they are unnecessary

Reply:
@CristoferMar There is a `Number.isInteger()` method to check that.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger